PL2/PL2-core/src/main/java/nl/tudelft/pl2016gr2/core/algorithms/GraphOrdererThread.java

```diff
@@ -0,0 +1,102 @@
+package nl.tudelft.pl2016gr2.core.algorithms;
+
+import nl.tudelft.pl2016gr2.model.AbstractNode;
+import nl.tudelft.pl2016gr2.model.GraphNodeOrder;
+import nl.tudelft.pl2016gr2.model.OriginalGraph;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Thread which can be used to order a graph.
+ *
+ * @author Faris
+ */
+public class GraphOrdererThread extends Thread {
+
+  private HashMap<Integer, GraphNodeOrder> orderedGraph;
+  private final OriginalGraph graph;
+
+  /**
+   * Construct a graph orderer thread.
+   *
+   * @param graph the graph to order.
+   */
+  public GraphOrdererThread(OriginalGraph graph) {
+    this.graph = graph;
+  }
+
+  /**
+   * Calculate the order of the nodes of the graph, so there are no backward edges. The given array
+   * list contains the nodes in order from left to right. Nodes which are at the same horizontal
+   * position have sequential positions in the array list and have the same value for their level
+   * field.
+   *
+   * @param graph the graph.
+   * @return the node order.
+   */
+  private static HashMap<Integer, GraphNodeOrder> calculateGraphOrder(OriginalGraph graph) {
+    HashMap<Integer, GraphNodeOrder> nodeOrder = new HashMap<>();
+    HashMap<Integer, Integer> reachedCount = new HashMap<>();
```

Comment:
This method is not so easy to understand. Maybe explain a little more in detail what you're doing. Especially the use of the reachedCount variable is not very clear at first sight.